OWML.ModHelper/Logging/ModSocketOutput.cs

```diff
@@ -20,34 +20,79 @@ public ModSocketOutput(IOwmlConfig config, IModLogger logger, IModManifest manif
             if (_socket == null)
             {
                 _port = config.SocketPort;
-                config.SocketPort = -1;
-                JsonHelper.SaveJsonObject(Constants.OwmlConfigFileName, config);
-                CreateSocket();
+                ConnectToSocket();
             }
         }
 
-        public override void WriteLine(string s)
+        [Obsolete]
+        public override void WriteLine(params object[] objects)
         {
-            Logger?.Log(s);
-            CallWriteCallback(Manifest, s);
-            var message = $"{Manifest.Name}{NameMessageSeparator}{s}";
-            InternalWriteLine(message);
+            var line = string.Join(" ", objects.Select(o => o.ToString()).ToArray());
+            MessageType type;
+            if (line.ToLower().Contains("error") || line.ToLower().Contains("exception"))
+            {
+                type = MessageType.Error;
+            }
+            else if (line.ToLower().Contains("warning") || line.ToLower().Contains("disabled"))
+            {
+                type = MessageType.Warning;
+            }
+            else if (line.ToLower().Contains("success"))
+            {
+                type = MessageType.Success;
+            }
+            else
+            {
+                type = MessageType.Message;
+            }
+            var senderType = GetCallingMethodName(new StackTrace());
+            WriteLine(senderType, type, line);
         }
 
-        public override void WriteLine(params object[] objects)
+
+        public override void WriteLine(string line, MessageType type = MessageType.Message)
         {
-            WriteLine(string.Join(" ", objects.Select(o => o.ToString()).ToArray()));
+            WriteLine(GetCallingMethodName(new StackTrace()), type, line);
+        }
+
+        private void WriteLine(string senderType, MessageType type, string line)
```

Comment:
the parameter order is inconsistent between the different methods.
I suggest: line, type, sender.

Reply:
fixed i think?